lib/sycamore/sycamore/tests/integration/connectors/opensearch/test_opensearch_read.py

```diff
@@ -153,51 +210,33 @@ def doc_reconstructor(index_name: str, doc_id: str) -> Document:
             .take_all()
         )
 
-        with OpenSearch(**TestOpenSearchRead.OS_CLIENT_ARGS) as os_client:
-            os_client.indices.refresh(TestOpenSearchRead.INDEX)
+        os_client.indices.refresh(TestOpenSearchRead.INDEX)
 
-        retrieved_docs = context.read.opensearch(
-            os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS, index_name=TestOpenSearchRead.INDEX
-        )
-        target_doc_id = original_docs[-1].doc_id if original_docs[-1].doc_id else ""
-        query = {"query": {"term": {"_id": target_doc_id}}}
-        query_docs = context.read.opensearch(
-            os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS, index_name=TestOpenSearchRead.INDEX, query=query
-        )
+        expected_count = len(original_docs)
+        actual_count = get_doc_count(os_client, TestOpenSearchRead.INDEX)
+        # refresh should have made all ingested docs immediately available for search
+        assert actual_count == expected_count, f"Expected {expected_count} documents, found {actual_count}"
 
-        retrieved_docs_reconstructed = context.read.opensearch(
-            os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS,
-            index_name=TestOpenSearchRead.INDEX,
-            reconstruct_document=True,
-            doc_reconstructor=doc_reconstructor,
+        retrieved_docs_reconstructed = (
+            context.read.opensearch(
+                os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS,
+                index_name=TestOpenSearchRead.INDEX,
+                reconstruct_document=True,
+                doc_reconstructor=OpenSearchDocumentReconstructor(TestOpenSearchRead.INDEX, doc_reconstructor),
+            )
+            .filter(lambda doc: doc.doc_id == "1")
```

Comment:
Why not read them all, sort and compare?  The filter seems unnecessary. You could drop all the docs except for 1 and the test would remain the same.

Reply:
ok